db/mongodb.go

```diff
@@ -247,3 +267,38 @@ func (m *MongoDB) CreateExtraIndexes(idxs []string) error {
 	slog.Info(fmt.Sprintf("%d index(es) successfully created in the collection %s", len(r), companyTableName))
 	return nil
 }
+
+func (m *MongoDB) Search(q Query) (string, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	coll := m.db.Collection(companyTableName)
+	f := bson.M{"json.uf": q.Uf}
+	if q.Cursor != "" {
+		f["json.cnpj"] = bson.M{"$gt": q.Cursor}
+	}
+	opts := options.Find().
+		SetSort(bson.D{{Key: "json.cnpj", Value: 1}}).
+		SetLimit(q.Limit)
+	c, err := coll.Find(ctx, f, opts)
+	if err != nil {
+		return "", fmt.Errorf("error running query %#v: %w", q.Uf, err)
+	}
+	defer c.Close(ctx)
+	var r []record
+	if err := c.All(ctx, &r); err != nil {
+		return "", fmt.Errorf("error decoding results: %w", err)
+	}
+	var p page
+	if len(r) > 0 {
+		var cs []transform.Company
+		for _, k := range r {
+			cs = append(cs, k.Json)
+		}
+		p = newPage(cs)
+	}
```

Comment:
Esse bloco pode ser simplificado pois muito dessa lógica foi encapsulada no `newPage`

```suggestion
        var cs []transform.Company
        for _, c := range r {
	        cs = append(cs, c.Json)
        }
        p := newPage(cs)
```